sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -454,6 +492,322 @@ pi_result OCL(piEnqueueMemBufferMap)(
+                        CL_KERNEL_EXEC_INFO_INDIRECT_SHARED_ACCESS_INTEL,
+                        sizeof(cl_bool), &TrueVal);
+  }
+  return PI_SUCCESS;
+}
+
+// USM memset API
+pi_result OCL(piEnqueueMemset)(pi_queue queue, void *ptr, pi_int32 value,
+                               size_t count, pi_uint32 num_events_in_waitlist,
+                               const pi_event *events_waitlist,
+                               pi_event *event) {
+
+  // Have to look up the context from the kernel
+  cl_context CLContext;
+  cl_int CLErr =
+      clGetCommandQueueInfo(cast<cl_command_queue>(queue), CL_QUEUE_CONTEXT,
+                            sizeof(cl_context), &CLContext, nullptr);
+  if (CLErr != CL_SUCCESS) {
+    return cast<pi_result>(CLErr);
+  }
+
+  clEnqueueMemsetINTEL_fn FuncPtr;
+  pi_result RetVal = getExtFuncFromContext<clEnqueueMemsetINTEL_fn>(
+      cast<pi_context>(CLContext), "clEnqueueMemsetINTEL", &FuncPtr);
+
+  if (RetVal != PI_SUCCESS)
+    return RetVal;
+
+  return cast<pi_result>(FuncPtr(
+      cast<cl_command_queue>(queue), ptr, value, count, num_events_in_waitlist,
+      cast<const cl_event *>(events_waitlist), cast<cl_event *>(event)));
+}
+
+// USM routine to copy data between host and device
+pi_result OCL(piEnqueueMemcpy)(pi_queue queue, pi_bool blocking, void *dst_ptr,
+                               const void *src_ptr, pi_int32 size,
+                               pi_uint32 num_events_in_waitlist,
+                               const pi_event *events_waitlist,
+                               pi_event *event) {
+
+  // Have to look up the context from the kernel
+  cl_context CLContext;
+  cl_int CLErr =
+      clGetCommandQueueInfo(cast<cl_command_queue>(queue), CL_QUEUE_CONTEXT,
+                            sizeof(cl_context), &CLContext, nullptr);
+  if (CLErr != CL_SUCCESS) {
+    return cast<pi_result>(CLErr);
+  }
+
+  clEnqueueMemcpyINTEL_fn FuncPtr;
+  pi_result RetVal = getExtFuncFromContext<clEnqueueMemcpyINTEL_fn>(
+      cast<pi_context>(CLContext), "clEnqueueMemcpyINTEL", &FuncPtr);
+
+  if (RetVal != PI_SUCCESS)
+    return RetVal;
+
+  return cast<pi_result>(FuncPtr(cast<cl_command_queue>(queue), blocking,
+                                 dst_ptr, src_ptr, size, num_events_in_waitlist,
+                                 cast<const cl_event *>(events_waitlist),
+                                 cast<cl_event *>(event)));
+}
+
+// Hint to migrate memory to the device
+pi_result OCL(piEnqueuePrefetch)(pi_queue queue, const void *ptr, size_t size,
+                                 pi_usm_migration_flags flags,
+                                 pi_uint32 num_events_in_waitlist,
+                                 const pi_event *events_waitlist,
+                                 pi_event *event) {
+
+  return cast<pi_result>(clEnqueueMarkerWithWaitList(
+      cast<cl_command_queue>(queue), num_events_in_waitlist,
+      cast<const cl_event *>(events_waitlist), cast<cl_event *>(event)));
+
+  /*
```

Comment:
why is this commented out? remove it?

Reply:
Sorry, forgot the comment that the commented out version will be used once devices actually support this.